mlir/lib/Dialect/Vector/IR/VectorOps.cpp

```diff
@@ -5181,6 +5181,19 @@ std::optional<SmallVector<int64_t, 4>> GatherOp::getShapeForUnroll() {
   return llvm::to_vector<4>(getVectorType().getShape());
 }
 
+static LogicalResult isContiguousIndices(Value val) {
+  auto vecType = dyn_cast<VectorType>(val.getType());
+  if (!vecType || vecType.getRank() != 1 || vecType.isScalable())
+    return failure();
+
+  DenseIntElementsAttr elements;
+  if (!matchPattern(val, m_Constant(&elements)))
+    return failure();
+
+  return success(
+      llvm::equal(elements, llvm::seq<int64_t>(0, vecType.getNumElements())));
+}
```

Comment:
@banach-space, is there a common utility that we can use here and for the extract op in the Linalg vectorizer?

Reply:
Not yet  - the vectorizer looks at the scalar indices _before_ vectorization. However, this patch make me think that we could do better 🤔 Let me look into this!

Reply:
@Hardcode84 Would you have some linalg examples that vectorize into these contiguous gathers? That would be helpful, but no worries if `Vector` is your actual starting point here.